react/SelectBox/Readme.md

```diff
@@ -327,13 +327,13 @@ const ExampleDialog = ({ open, onClose }) => {
   // transition. Otherwise the position for the selectbox is computed
   // wrongly. See the `transitionDuration` prop.
   return (
-    <Dialog 
-      open={open} 
-      onClose={onClose} 
+    <Dialog
+      open={open}
+      onClose={onClose}
       title='Ada Lovelace'
       transitionDuration={0}
       content={
-        <Typography variant='body1'>
+        <>
```

Comment:
What were the errors? Is it legit to have text without being wrapped inside a Typo component?

Reply:
quelque chose du genre "pas possible d'avoir un `<p>` descendant d'un `<div>`" donc non je ne pense pas que ce soit legit

Reply:
`<p>` inside `<div>` is totally right. The contrary is not. 

The `content` is inserted inside a `DialogContent`: https://github.com/cozy/cozy-ui/blob/master/react/CozyDialogs/Dialog.jsx#L46-L48

DialogContent comes from MUI: https://github.com/mui-org/material-ui/blob/v4.x/packages/material-ui/src/DialogContent/DialogContent.js#L26-L42 
So you can see we can pass what we want. It's also specified in the documentation: https://v4.mui.com/api/dialog-content/#props 

I don't understand why we can't use a Typography here.


I think the issue is that we use a `SelectBox` inside a typography component. This is the real issue. 

I persist to say we should not have any "string" without being wrapped in a Typography component

Reply:
C'est l'inverse c'est `<div> cannot appear as a descendant of <p>`

Reply:
I still don't understand why you say: 
> Le content d'un Dialog est censé venir avec son style

And then 
> par défaut on n'a pas besoin d'un Typography

This is precisely because the content of THIS dialog is a "string" that we need to wrap it inside a "Typography" since this is how we apply a style to the a string. Here the content is a string, we have to use Typography to have its own style.

Again, we should not have any string without being wrapped inside a Typography component. We should not have any exemple with a string without a typo component, we should not have any span with css's classes dealing with text's style and so on. Typography all the way. 

> Dans cette exemple il n'a pas de valeur ajouté

Let's see what happen when we used inverted or dark theme ;)

Reply:
Oui ok pour ne pas avoir de string sans Typography.

Ici on est dans un cozyDialog (et quand bien même une Dialog marcherait), "content" a le bon style. Sans typography, ça réagit correctement selon le thème.